libs/cardano-data/src/Data/OMap/Strict.hs

```diff
@@ -419,12 +423,15 @@ instance (Typeable k, EncCBOR v, Ord k) => EncCBOR (OMap k v) where
   encCBOR omap = encodeStrictSeq encCBOR (toStrictSeq omap)
 
 instance (Typeable k, HasOKey k v, DecCBOR v, Eq v) => DecCBOR (OMap k v) where
-  decCBOR =
-    decodeListLikeEnforceNoDuplicates
-      decodeListLenOrIndef
-      (flip snoc)
-      (\omap -> (size omap, omap))
-      decCBOR
+  decCBOR = decodeOMap decCBOR
+
+decodeOMap :: HasOKey k v => Decoder s v -> Decoder s (OMap k v)
```

Comment:
Looks like this ended up not being used, but I guess it doesn't harm to have it anyway.

Reply:
Yes, I was using it initially, but it doesn't allow adding to the sharing state for each individual proposal, so I abandon the idea, but this function was retained for the sake of consistency with decoders for other data structures (eg, `decodeMap`, `decodeSet`, etc)